docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,345 @@
+2. ocicrypt-rs calls the keyprovider grpc service provided by the attestation-agent and sends the annotation packet to the attestation-agent.
+3. The attestation-agent performs HW-TEE remote attestation with the relying KBS deployed by the remote user. After passing the attestation, it interacts with KBS through the encrypted channel, and decrypts the PrivateLayerBlockCipherOptions using the user's public key in the KBS.
+4. ocicrypt-rs obtains the decrypted PrivateLayerBlockCipherOptions from the attestation-agent, reads the symmetric key used to decrypt the container image, and completes the decryption of the container image layer.
+
+## Attestation Agent
+
+The attestation-agent is an indispensable core component in the confidential containers architecture. It undertakes the trust distribution function of the confidential container. In the process of signature verification and decryption of the protected image, the attestation-agent serves as the source of the user's confidential information and the decryptor of the PrivateLayerBlockCipherOptions. As mentioned above, it provides two gRPC services for image-rs and ocicrypt-rs to support the above two functions.
+
+### GetResource service
+
+The GetResource grpc service interface provided by the attestation agent is used to obtain various confidential resources from the relying party (KBS). In the process of protected image deployment, it is usually used to obtain policiy.json and sigstore config file. 
+
+Attestation-agent performs HW-TEE attestation to the KBS, establish an encrypted channel after negotiating the key, and download the required confidential resources through the encrypted channel. The GetResource grpc service interface protobuf of the attachment agent is defined as follows:
+
+```protobuf
+message getResourceRequest {
+    string KbcName = 1;
+    string KbsUri = 2;
+    string ResourceDescription = 3;
+}
+
+message getResourceResponse {
+    bytes resource = 1;
+}
+
+service GetResource {
+    rpc GetResource(getResourceRequest) returns (getResourceResponse) {};
+}
+```
+
+ResourceDescription is a JSON string used to describe resources, and its format is defined as follows:
+
+```json
+{
+    "name":"example_resource",
+    "informations":{
+        "index":"example_index",
+        "extra_param":[]
+    },
+}
+```
+
+In the design of confidential containers V1, when get policy.json through GetResource grpc endpoint, the contents of ResourceDescription are as follows:
+
+```json
+{
+    "name":"policy.json",
+    "informations":{
+        "index":"",
+        "extra_param":[]
+    },
+}
+```
+
+When get sigstore config file through GetResource grpc endpoint, the contents of ResourceDescription are as follows:
+
+```json
+{
+    "name":"sigstore_config",
+    "informations":{
+        "index":"",
+        "extra_param":[]
+    },
+}
+```
+
```

Comment:
This interface seems fine, but it is notable that none of the examples use anything in the `information` field. It's probably good to keep things flexible, though, as we should be able to use this to provide anything to the guest in the future.

One note is that the `offline_sev_kbc` is limited by the size of the injected secret. We may run into issues when we start trying to send in a bunch of stuff. As mentioned, we should think about fallbacks if certain KBC/KBS doesn't support certain endpoints. We can make an online KBS (even with SEV(-ES)) that won't have the same limitation.

We should also think about how we plan to implement this in the short-term. Maybe we should have a PR where we add these endpoints as a stub to every existing KBC and then the individual maintainers can go through and implement how they would like.

Reply:
As mentioned above, obtaining resources from AA will only be triggered when image-rs does not find the configuration files in the local (rootfs) directory. For `offline_sev_kbc`, these resources can be preset in rootfs when making boot image, or provided by sev KBC in any special way (for example, use the simple policy of hard coding in KBC, or, as you mentioned, build an additional online KBC, etc.).

About the implementation of this interface of AA, I will start coding next week:)